xmppserver/src/main/java/org/jivesoftware/openfire/muc/MUCOccupant.java

```diff
@@ -40,6 +40,7 @@
 import java.io.ObjectInput;
 import java.io.ObjectOutput;
 import java.util.HashSet;
+import java.util.List;
 import java.util.Set;
 import java.util.stream.Collectors;
 
```

Comment:
What about the other 2 remove()'s in this file?

```
newPresence.getElement().remove(element);
// and
            if (mucUser != null) {
                // Remove any previous extendedInformation, then re-add it.
                presence.getElement().remove(mucUser);
            }
```
Seems like maybe those should remove all and not just 1 too ?

Reply:
also the same in `MultiUserChatServiceImpl.java`

Reply:
Those should be tackled too, I agree. I've created a separate PR for those: https://github.com/igniterealtime/Openfire/pull/2761